### PR TITLE
Feature/127339643

### PR DESCRIPTION
**Pull request by monasaur**
- [x] Can add subject
- [ ] Cannot add student in add subject
- [x] Fixes Login
- [x] Removes Student and Admin's dashboard

### CODE REVIEW

File: app/controllers/grades_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class GradesController < ApplicationController
+
+  def index
+  	@grade = Grade.all
```
Comment (goldendragon1988) [inline app/controllers/grades_controller.rb]: spacing.
Also please use the plural form for "@grade" for this query is a collection.

File: app/controllers/grades_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def new	
+  end
+
+  def create
+    @grade = Grade.new(grade_params)
+ 
+    @grade.save
+    redirect_to grade_path(@grade)
```
Comment (goldendragon1988) [inline app/controllers/grades_controller.rb]: what happen if the grade is not created?

File: app/controllers/students_controller.rb [hunk 3]
```diff
@@ -1,29 +1,20 @@
...
-  end
-
-  def show
-  end
-
-  def new
-    @student = Student.new
-  end
+def show
+    @student = Student.find(params[:id])
```
Comment (goldendragon1988) [inline app/controllers/students_controller.rb]: spacing

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -1,29 +1,20 @@
...
 
-  private
-    
-    def set_student
-      @student = Student.find(params[:id])
-    end
+def create
+  @student = Student.new(student_params)
+ 
+  @student.save
```
Comment (goldendragon1988) [inline app/controllers/students_controller.rb]: same as above for failed creation.

File: app/controllers/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class SubjectsController < ApplicationController
+
+ def index
```
Comment (goldendragon1988) [inline app/controllers/subjects_controller.rb]: spacing

File: app/controllers/subjects_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @subject = Subject.find(params[:id])
+  end
+
+  def new
+  end
+
+  def create
+    @subject = Subject.new(subject_params)
+ 
+    @subject.save
```
Comment (goldendragon1988) [inline app/controllers/subjects_controller.rb]: failed creation.

File: app/controllers/users/sessions_controller.rb [hunk 7]
```diff
@@ -2,8 +2,8 @@ class Users::SessionsController < Devise::SessionsController
 
 private
   def after_sign_in_path_for(resource)
-  	if current_user.role == "Admin"
-        admin_dashboard_path
+  	if current_user.role == "Professor"
+       dashboard_path
       elsif current_user.role == "Professor"
         dashboard_path
       else
```
Comment (goldendragon1988) [inline app/controllers/users/sessions_controller.rb]: please check, the other condition is also a professor.

File: app/controllers/grades_controller.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+private
+  def grade_params
+    params.require(:grades).permit(:squiz1, :squiz2, :lquiz1, :lquiz2, :seatwork1, :seatwork2, :seatwork3, :project)
```
Comment (goldendragon1988) [inline app/controllers/grades_controller.rb]: refactor this one. Accept one quiz at a time. or used an array in accepting the quiz.

File: app/views/landing_page/dashboard.html.haml [hunk 9]
```diff
@@ -1,33 +1,32 @@
...
+    = link_to 'Change Password',  edit_user_registration_path
+    %div
+      = link_to 'Sign out', destroy_user_session_path, method: :delete
+      %hr/
+      %center
+        %table{:border => "5"}
+          %tr
+            %th Student Number
+            %th Student Name
+            %th= link_to 'Short Quiz 1'
```
Comment (goldendragon1988) [inline app/views/landing_page/dashboard.html.haml]: have a single link only to "Quiz" and just add a selection for the type of quiz.

File: app/views/students/new.html.haml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+%h1 Student's Grade
+%br/
```
Comment (goldendragon1988) [inline app/views/students/new.html.haml]: delete

File: app/views/subjects/new.html.haml [hunk 11]
```diff
@@ -0,0 +1,22 @@
+%h1 Subject
+%br/
```
Comment (goldendragon1988) [inline app/views/subjects/new.html.haml]: delete

File: db/migrate/20160801090619_create_grades.rb [hunk 12]
```diff
@@ -0,0 +1,16 @@
+class CreateGrades < ActiveRecord::Migration
+  def change
+    create_table :grades do |t|
+    	t.integer :squiz1
```
Comment (goldendragon1988) [inline db/migrate/20160801090619_create_grades.rb]: create an association for this.

File: db/migrate/20160802075637_remove_middle_name_to_user.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+class RemoveMiddleNameToUser < ActiveRecord::Migration
+  def change
+  	remove_column :users, :middle_name
```
Comment (goldendragon1988) [inline db/migrate/20160802075637_remove_middle_name_to_user.rb]: spacing.

File: db/migrate/20160803081612_drop_subjects.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+class DropSubjects < ActiveRecord::Migration
+  def change
+  	drop_table :subjects
```
Comment (goldendragon1988) [inline db/migrate/20160803081612_drop_subjects.rb]: spacing.

File: app/controllers/users/sessions_controller.rb [hunk 15]
```diff
@@ -2,12 +2,7 @@ class Users::SessionsController < Devise::SessionsController
...
 private
   def after_sign_in_path_for(resource)
-  	if current_user.role == "Admin"
-        admin_dashboard_path
-      elsif current_user.role == "Professor"
-        dashboard_path
-      else 
-      	student_dashboard_path
-    end
+  	if current_user.role == "Professor"
```
Comment (edos4) [inline app/controllers/users/sessions_controller.rb]: Remove if statement when there are no other roles.

File: app/views/devise/registrations/new.html.haml [hunk 16]
```diff
@@ -19,7 +19,7 @@
   .field
     = f.label :role
   %div
-    = f.select :role, options_for_select(["Admin", "Professor"], f.object.role)
+    = f.select :role, options_for_select(["Professor"], f.object.role)
```
Comment (edos4) [inline app/views/devise/registrations/new.html.haml]: There are no roles other than professor. Remove this.

**Pull request merged**